### PR TITLE
JUnit5 :: unit, extraction, id, and functional tests

### PR DESCRIPTION
**Pull request by dev-mlb**

### CODE REVIEW

File: src/test/java/emissary/output/roller/JournaledCoalescerTest.java [hunk 1]
```diff
@@ -147,11 +146,10 @@ public void testRollOrphanedFiles() throws Exception {
 
         // verify
         Path bud1destination = targetBUDPath.resolve(BUD1_NAME);
-        assertThat(Files.exists(bud1destination), equalTo(true));
+        assertTrue(Files.exists(bud1destination));
         List<String> fileResults = Files.readAllLines(bud1destination, Charset.defaultCharset());
-        assertThat("Expected 4 lines in " + bud1destination, fileResults.size(), equalTo(4));
-        assertThat(fileResults, containsInRelativeOrder(BUD1_LINES.get(0), BUD1_LINES.get(1)));
-        assertThat(fileResults, containsInRelativeOrder(BUD2_LINES.get(0), BUD2_LINES.get(1)));
+        assertEquals(4, fileResults.size());
```
[the remark this answers is not in the document]
Comment (dev-mlb) [reply src/test/java/emissary/output/roller/JournaledCoalescerTest.java]: not part of this pr, was in a previous one

File: src/main/java/emissary/test/core/ExtractionTest.java [hunk 2]
```diff
@@ -223,10 +213,9 @@ protected void checkAnswers(Element el, IBaseDataObject payload, List<IBaseDataO
         // Check specified nometa
         for (Element meta : el.getChildren("nometa")) {
             String key = meta.getChildTextTrim("name");
-            assertTrue(
```
[the remark this answers is not in the document]
Comment (dev-mlb) [reply src/main/java/emissary/test/core/ExtractionTest.java:226]: yup good catch

File: src/test/java/emissary/output/roller/JournaledCoalescerTest.java [hunk 3]
```diff
@@ -147,11 +146,10 @@ public void testRollOrphanedFiles() throws Exception {
...
+        assertTrue(fileResults.containsAll(Arrays.asList(BUD1_LINES.get(0), BUD1_LINES.get(1), BUD2_LINES.get(0), BUD2_LINES.get(1))));
```
[the remark this answers is not in the document]
Comment (dev-mlb) [reply src/test/java/emissary/output/roller/JournaledCoalescerTest.java]: not part of this pr, was in a previous one

File: src/test/java/emissary/output/roller/JournaledCoalescerTest.java [hunk 4]
```diff
@@ -204,13 +202,13 @@ public void testAddFilesWithRoll() throws Exception {
         }
         // verify
         Path bud1Destination = targetBUDPath.resolve(expectedPrefix1);
-        assertThat("Expected target to exist " + bud1Destination.toString(), Files.exists(bud1Destination), equalTo(true));
-        assertThat(bud1Destination.getFileName().toString(), not(equalTo(expectedPrefix2)));
-        assertThat(bud1Destination.getParent(), equalTo(targetBUDPath));
-        assertThat(bud1Destination.getFileName().toString(), equalTo(expectedPrefix1));
-        assertThat(Files.readAllLines(bud1Destination, Charset.defaultCharset()), equalTo(BUD1_LINES));
+        assertTrue(Files.exists(bud1Destination), "Expected target to exist " + bud1Destination);
+        assertNotEquals(bud1Destination.getFileName().toString(), expectedPrefix2);
```
[the remark this answers is not in the document]
Comment (dev-mlb) [reply src/test/java/emissary/output/roller/JournaledCoalescerTest.java]: not part of this pr, was in a previous one

File: src/test/java/emissary/output/roller/JournaledCoalescerTest.java [hunk 5]
```diff
@@ -204,13 +202,13 @@ public void testAddFilesWithRoll() throws Exception {
...
+        assertEquals(targetBUDPath, bud1Destination.getParent());
+        assertEquals(expectedPrefix1, bud1Destination.getFileName().toString());
+        assertEquals(BUD1_LINES, Files.readAllLines(bud1Destination, Charset.defaultCharset()));
 
-        assertThat(expectedPrefix1, not(equalTo(expectedPrefix2)));
+        assertNotEquals(expectedPrefix1, expectedPrefix2);
```
[the remark this answers is not in the document]
Comment (dev-mlb) [reply src/test/java/emissary/output/roller/JournaledCoalescerTest.java]: not part of this pr, was in a previous one

**Comment by dev-mlb** [comment]
To use the test jar --
```
     <dependency>
        <groupId>emissary</groupId>
        <artifactId>emissary</artifactId>
        <version>${emissary.version}</version>
        <classifier>tests</classifier>
        <type>test-jar</type>
        <scope>test</scope>
      </dependency>
```

**Review by arp-0984 — APPROVED** [review]

**Review by jdcove2 — APPROVED** [review]

**Comment by dev-mlb** [comment]
Breaking this up into separate PRs see #278 #279 #300

**Comment by jpdahlke** [comment]
> Breaking this up into separate PRs see #278 #279 #300 #301

These are all merged now.

**Pull request closed without merging**